app/models/emoji.rb

```diff
@@ -48,19 +43,30 @@ def self.exists?(name)
   end
 
   def self.[](name)
-    Emoji.custom.detect { |e| e.name == name }
+    name = name.delete_prefix(':').delete_suffix(':')
+    is_toned = name.match?(/.+:t[1-6]/)
+    normalized_name = name.gsub(/(.+):t[1-6]/, '\1')
+
+    Emoji.all.detect do |e|
+      e.name == normalized_name &&
+      (!is_toned || is_toned && e.tonable)
```

Comment:
Depending on the precedence of the `||` and the `&&` operators, this might not do what you want.

It's always better to use parentheses when mixing them.

```suggestion
      (!is_toned || (is_toned && e.tonable))
```